src/optimalcontrol/include/iDynTree/DynamicalSystem.h

```diff
@@ -50,9 +50,18 @@ namespace optimalcontrol {
                               double time,
                               VectorDynSize& stateDynamics) = 0;
 
-        virtual const VectorDynSize& initialState() const = 0;
+        virtual bool setControlInput(const VectorDynSize &control);
+
+        virtual const VectorDynSize& controlInput() const;
+
+        virtual double controlInput(unsigned int index) const;
```

Comment:
Not a big problem, but I would use `size_t` in place of `unsigned int` for indexes .

Reply:
I was keeping consistency with ``VectorDynSize``. Anyway I can change it if you prefer.

Reply:
I see, my fault then. : D 
Let's leave like that and eventually we will change  it also in `VectorDynSize` .